components/profile-controller/controllers/profile_controller.go

```diff
@@ -524,6 +527,15 @@ func (r *ProfileReconciler) getAuthorizationPolicy(profileIns *profilev1.Profile
 			},
 		},
 	}
+
+	// In ambient mode, we still use selector but target the waypoint workload
+	// TODO: Once Istio supports targetRef in AuthorizationPolicy, update this
+	if r.ServiceMeshMode == "istio-ambient" {
+		// For now, keep the selector-based approach for ambient mode
+		// The waypoint will be created separately and policies will apply to namespace workloads
```

Comment:
Istio in ambient mode already supports the use of targetRef. That's how it can be bound to all traffic passing through the whole namespace https://github.com/orfeas-k/kubeflow-istio-ambient-demo/blob/main/use-case-c-programmatic-access-using-k8s-token/ns-owner-access-istio-ap-l7-waypoint.yaml#L38-L41.